Cleanup / micro-optimize RenderLayers accumulateOffsetTowardsAncestor helper

nikolaszimmermann opened the pull request:
#### ae09dde3f987f7d22a753377fb2a371d84da7549
<pre>
Cleanup / micro-optimize RenderLayers accumulateOffsetTowardsAncestor helper
<a href="https://bugs.webkit.org/show_bug.cgi?id=245395">https://bugs.webkit.org/show_bug.cgi?id=245395</a>

Reviewed by Rob Buis.

Avoid unnecessary LayoutPoint -&gt; LayoutSize conversions (e.g. when applying
scroll offset, or when moving &apos;location&apos; by the result of &apos;localToAbsolute&apos;).

Use auto in a few more places, where it helps readability.

Covered by existing tests.

* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::accumulateOffsetTowardsAncestor):

Canonical link: <a href="https://commits.webkit.org/254669@main">https://commits.webkit.org/254669@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/827756fb76d9144ad8e6a168fb4f3dc71ecc3322

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/89772 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/34323 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/20475 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/99106 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/155923 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/93780 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/32817 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/28269 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/82130 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/93456 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/95419 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/26077 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/76609 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/26015 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80954 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80807 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/69010 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/30567 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14891 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/30316 "Built successfully") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15828 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/embedded-content/the-iframe-element/sandbox-toggle-in-inactive-document-crash.html (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3280 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/33767 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38777 "Found 1 new test failure: imported/w3c/web-platform-tests/html/semantics/embedded-content/the-iframe-element/sandbox-toggle-in-inactive-document-crash.html (failure)") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/32478 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34916 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

rwlbuis commented on Source/WebCore/rendering/RenderLayer.cpp:
```diff
@@ -2454,10 +2452,10 @@ static inline const RenderLayer* accumulateOffsetTowardsAncestor(const RenderLay
+            auto* positionedAncestor = parentLayer->enclosingAncestorForPosition(position);
+            auto thisCoords = layer->offsetFromAncestor(positionedAncestor);
+            auto ancestorCoords = ancestorLayer->offsetFromAncestor(positionedAncestor);
+            location.move(thisCoords - ancestorCoords);
```
Is this really more efficient?

nikolaszimmermann replied:
No, move and operator+= are equally fast. I just find out looks awkward to have places with `location.moveBy(...)` (with no alternative, since operator += is only defined to add a LayoutSize, not a LayoutPoint) and others with use `location += ...`. WDYT?

rwlbuis replied:
I see your point, agreed.

rwlbuis reviewed (approved):
LGTM.

nikolaszimmermann committed:
Cleanup / micro-optimize RenderLayers accumulateOffsetTowardsAncestor helper
https://bugs.webkit.org/show_bug.cgi?id=245395

Reviewed by Rob Buis.

Avoid unnecessary LayoutPoint -> LayoutSize conversions (e.g. when applying
scroll offset, or when moving 'location' by the result of 'localToAbsolute').

Use auto in a few more places, where it helps readability.

Covered by existing tests.

* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::accumulateOffsetTowardsAncestor):

Canonical link: https://commits.webkit.org/254669@main
